error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/JUnitClassModifiersTest.java

```diff
@@ -0,0 +1,129 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class JUnitClassModifiersTest {
+  private final CompilationTestHelper compilationTestHelper =
+      CompilationTestHelper.newInstance(JUnitClassModifiers.class, getClass());
+  private final BugCheckerRefactoringTestHelper refactoringTestHelper =
+      BugCheckerRefactoringTestHelper.newInstance(JUnitClassModifiers.class, getClass());
+
+  @Test
+  void identification() {
+    compilationTestHelper
+        .addSourceLines(
+            "Container.java",
+            "import org.junit.jupiter.api.Test;",
+            "import org.junit.jupiter.params.ParameterizedTest;",
+            "import org.springframework.boot.test.context.TestConfiguration;",
+            "import org.springframework.context.annotation.Configuration;",
+            "",
+            "class Container {",
+            "  final class FinalAndPackagePrivate {",
+            "    @Test",
+            "    void foo() {}",
+            "  }",
+            "",
+            "  public abstract class Abstract {",
+            "    @Test",
+            "    void foo() {}",
+            "  }",
+            "",
+            "  @Configuration",
+            "  class WithConfigurationAnnotation {",
+            "    @Test",
+            "    void foo() {}",
+            "  }",
+            "",
+            "  @TestConfiguration",
+            "  class WithConfigurationMetaAnnotation {",
+            "    @Test",
+            "    void foo() {}",
+            "  }",
+            "",
+            "  // BUG: Diagnostic contains:",
+            "  private final class Private {",
+            "    @Test",
+            "    void foo() {}",
+            "  }",
+            "",
+            "  // BUG: Diagnostic contains:",
+            "  protected final class Protected {",
+            "    @Test",
+            "    void foo() {}",
+            "  }",
+            "",
+            "  // BUG: Diagnostic contains:",
+            "  public final class Public {",
+            "    @Test",
+            "    void foo() {}",
+            "  }",
+            "",
+            "  // BUG: Diagnostic contains:",
+            "  class NonFinal {",
+            "    @Test",
+            "    void foo() {}",
+            "  }",
+            "",
+            "  // BUG: Diagnostic contains:",
+            "  class NonFinalWithCustomTestMethod {",
+            "    @ParameterizedTest",
+            "    void foo() {}",
+            "  }",
```

Comment:
I feel like these two cases are misleading: IIUC, the latter doesn't fail due to the "custom test method" but solely because it's non-final. Did I get this right?

Shouldn't we add a _positive_ test case for test methods annotated with `@ParameterizedTest`? Though one could argue this is already covered by the test cases for `MoreJUnitMatchers#TEST_METHOD` in `MoreJUnitMatchersTest`.

Reply:
> Did I get this right?

Yes :)

> Shouldn't we add a _positive_ test case for test methods annotated with `@ParameterizedTest`?

Can do :)